include/mgclient.hpp

```diff
@@ -50,6 +50,18 @@ class Client {
   /// If there is nothing to fetch, `std::nullopt` is returned.
   std::optional<std::vector<Value>> FetchOne();
 
+  /// \brief Start a transaction.
+  /// \return true when the transaction was successfully started, false
+  /// otherwise.
+  bool BeginTransaction();
+
+  /// \brief End current transaction.
+  /// \param commitTransaction whether the transaction should be commited
+  /// or rollbacked.
+  /// \return true when the transaction was successfully ended, false
+  /// otherwise.
+  bool EndTransaction(bool commitTransaction);
```

Comment:
I thought about that and maybe I was too biased toward this solution because the implementation is basically the same for both options, the only difference being the message signature that is being sent. 
I agree that it is easier to use and understand by having separate methods, in C++ wrapper, and in the C API...